Initium-Odp/src/com/universeprojects/miniup/server/scripting/wrappers/Character.java

```diff
@@ -399,4 +402,64 @@ public Group getGroup()
 		
 		return characterGroup;
 	}
+	
+	/*################# KNOWLEDGE ###################*/
+	public Knowledge getKnowledgeByName(String name) {
+		CachedEntity toReturn = getKS().fetchKnowledgeByName(name);
+		if(toReturn == null) return null;
+		else return new Knowledge(toReturn, db);
+	}
+	
+	/**
+	 * Fetch all knowledge entities associated with this character.
+	 * @return
+	 */
+	public Knowledge[] getAllKnowledge(){
+		List<Knowledge> toReturn = new ArrayList<>();
+		for(CachedEntity ce : getKS().getAllKnowledge()) 
+			toReturn.add(new Knowledge(ce, db));
+		
+		return toReturn.toArray(new Knowledge[toReturn.size()]);
+	}
+	
+	/**
+	 * Fetchs the knowledge chain for an item. Creates it if it doesnt exist.
+	 * @param item
+	 * @return
+	 */
+	public Knowledge[] getKnowledgeForItem(Item item) {
+		return getKnowledgeForItem(item, true);
+	}
+	
+	/**
+	 * Fetchs the knowledge chain for an item.
+	 * @param item
+	 * @param createIfNoExist
+	 * @return
+	 */
+	public Knowledge[] getKnowledgeForItem(Item item, boolean createIfNoExist) {
+		List<Knowledge> toReturn = new ArrayList<>();
+		for(CachedEntity ce : getKS().getKnowledgeEntitiesFor(item.getEntity(), createIfNoExist)) 
+			toReturn.add(new Knowledge(ce, db));
+		
+		return toReturn.toArray(new Knowledge[toReturn.size()]);
+	}
+	
+	/**
+	 * Increase the knowledge for a particular item.
+	 * @param item
+	 * @param amount
+	 * @param percentageCap - what percent of the knowledge cap can you get from this increase?
+	 * @return
+	 */
+	public boolean increaseKnowledgeFor(Item item, int amount, int percentageCap) {
+		return getKS().increaseKnowledgeFor(item.getEntity(), amount, percentageCap);
+	}
+	
+	private ODPKnowledgeService getKS() {

```

Comment:
Lets not use contractions like this. DB is ok because people use DB in every day language. But KS is not immediately obvious what it's referring to.